src/idom/core/hooks.py

```diff
@@ -144,15 +140,15 @@ def use_effect(
 
     dependencies = _try_to_infer_closure_values(function, dependencies)
     memoize = use_memo(dependencies=dependencies)
-    last_clean_callback: Ref[Optional[_EffectCleanFunc]] = use_ref(None)
+    last_clean_callback: Ref[_EffectCleanFunc | None] = use_ref(None)
 
     def add_effect(function: _EffectApplyFunc) -> None:
         if not asyncio.iscoroutinefunction(function):
             sync_function = cast(_SyncEffectFunc, function)
         else:
             async_function = cast(_AsyncEffectFunc, function)
 
-            def sync_function() -> Optional[_EffectCleanFunc]:
+            def sync_function() -> _EffectCleanFunc | None:
```

Comment:
We might want to add `pyupgrade` as a precommit-ci hook to automagically handle these

Reply:
That would be great to have

Reply:
I have a pretty exhaustive "zero effort auto fixes" precommit-ci config in Conreq's App store branch, if you want to take a peek.

https://github.com/Archmonger/Conreq/blob/d3740f4b7a340491c886dd114b7f2f40c398c4a1/.pre-commit-config.yaml

Reply:
Feel free to make a PR for this. I don't have time to check this out at the moment.

Reply:
Sure, I'll PR after v1.